### ISSUE TITLE
Seg. Fault starting simple shell

### ISSUE THREAD

**Issue by ricardomv** [greenisland/greenisland#183]
Can't start the default shell on any platform (tested eglfs and xcb) on 0.7.90.

xrandr
```
Screen 0: minimum 8 x 8, current 1920 x 1080, maximum 32767 x 32767
eDP1 connected primary 1920x1080+0+0 (normal left inverted right x axis y axis) 294mm x 165mm
   1920x1080     60.02*+
   1400x1050     59.98  
   1600x900      60.00  
   1280x1024     60.02  
   1280x960      60.00  
   1368x768      60.00  
   1280x720      60.00  
   1024x768      60.00  
   1024x576      60.00  
   960x540       60.00  
   800x600       60.32    56.25  
   864x486       60.00  
   640x480       59.94  
   720x405       60.00  
   640x360       60.00  
DP1 disconnected (normal left inverted right x axis y axis)
HDMI1 disconnected (normal left inverted right x axis y axis)
VIRTUAL1 disconnected (normal left inverted right x axis y axis)
```
GDB backtrace:

http://pastebin.com/hBvQ95zi

**Comment by ricardomv** [comment]
Tested version 0.7.1 and it works fine.

**Comment by iBeliever** [comment]
I'm getting the same error as well.

**Comment by ricardomv** [comment]
running git bissect looks like the first bad commit is c7ee3519

**Comment by ricardomv** [comment]
This can be closed, 7032bd8 seems to fix the issue.

**Comment by plfiorini** [comment]
Ok Thanks
